### ISSUE TITLE
Uncaught Error: spawn ENOTDIR

### ISSUE THREAD

**Issue by folour** [atom/ide-php#37]
1. Set socket connection type
2. Restart atom

**Atom**: 1.21.0-beta0 x64
**Electron**: 1.6.9
**OS**: Ubuntu 16.04.3
**Thrown From**: [ide-php](https://github.com/atom/ide-php) package 0.6.6

### Stack Trace

Uncaught Error: spawn ENOTDIR

```
At internal/child_process.js:313

Error: spawn ENOTDIR
    at exports._errnoException (util.js:1022:11)
    at ChildProcess.spawn (internal/child_process.js:313:11)
    at Object.exports.spawn (child_process.js:399:9)
    at PHPLanguageClient.spawnServer (/packages/ide-php/lib/main.js:75:29)
    at Server.server.listen (/packages/ide-php/lib/main.js:57:29)
    at Object.onceWrapper (events.js:290:19)
    at emitNone (events.js:86:13)
    at Server.emit (events.js:185:7)
    at emitListeningNT (net.js:1288:10)
    at _combinedTickCallback (internal/process/next_tick.js:71:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

### Commands

```
```

### Non-Core Packages

```
atom-ide-ui 0.3.3 
ide-php 0.6.6 
```

**Comment by damieng** [comment]
Can you post the contents of the output from the devtools window?
Also can you check your your ~/.atom/packages/ide-php folder and see if it contains a "vendor" folder?

**Comment by sr1994lu** [comment]
i have same problem.
i have Dir as "vendor".

**Comment by damieng** [comment]
Can you post a listing of the contents of the vendor folder?  i.e.

```ls ~/.atom/packages/ide-php/vendor```

**Comment by sr1994lu** [comment]
* autoload.php
* felixfbecker
* microsoft
* psr
* squizlabs
* bin
* jetbrains
* netresearch
* sabre
* symfony
* composer
* justinrainbow
* phpdocumentor
* seld
* webmozart

**Comment by sr1994lu** [comment]
and Confused about PHP Path

* "/Users/iam/.anyenv/envs/phpenv/versions/7.2.0RC1/"
* "/Users/iam/.anyenv/envs/phpenv/versions/7.2.0RC1/bin/php"

Which one should I choose?

**Comment by damieng** [comment]
It should point to a 'php' binary rather than the folder.

**Comment by sr1994lu** [comment]
OK.

i try to set 2 PHP runtimes.
* 7.1.9
* 7.2.0RC1

all failed.

Incidentally,
* mem limit set 2G.
* socket set auto.

**Comment by sr1994lu** [comment]
```
Uncaught (in promise) Error: spawn ENOTDIR
    at exports._errnoException (util.js:1022:11)
    at ChildProcess.spawn (internal/child_process.js:313:11)
    at Object.exports.spawn (child_process.js:399:9)
    at Promise (/Users/sr1994lu/.atom/packages/ide-php/lib/main.js:48:31)
    at PHPLanguageClient.checkPHPVersion (/Users/sr1994lu/.atom/packages/ide-php/lib/main.js:47:12)
    at PHPLanguageClient.spawnServer (/Users/sr1994lu/.atom/packages/ide-php/lib/main.js:105:17)
    at getOrInstallServer.then (/Users/sr1994lu/.atom/packages/ide-php/lib/main.js:26:83)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Comment by damieng** [comment]
What happens if you type this in a Terminal window?

```/Users/iam/.anyenv/envs/phpenv/versions/7.2.0RC1/bin/php --version``

**Comment by sr1994lu** [comment]
@damieng here
```
PHP 7.2.0RC1 (cli) (built: Sep 17 2017 03:27:21) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.2.0-dev, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.2.0RC1, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.6.0-dev, Copyright (c) 2002-2017, by Derick Rethans
```

**Comment by bstrahija** [comment]
Doesn't work at all for me:

```
Uncaught (in promise) Error: spawn ENOTDIR
    at exports._errnoException (util.js:1022)
    at ChildProcess.spawn (internal/child_process.js:313)
    at Object.exports.spawn (child_process.js:399)
    at Promise (/Volumes/Superhero/.atom/packages/ide-php/lib/main.js:48)
    at PHPLanguageClient.checkPHPVersion (/Volumes/Superhero/.atom/packages/ide-php/lib/main.js:47)
    at PHPLanguageClient.spawnServer (/Volumes/Superhero/.atom/packages/ide-php/lib/main.js:105)
    at getOrInstallServer.then (/Volumes/Superhero/.atom/packages/ide-php/lib/main.js:26)
```

PHP versions and paths:

```
php -v
PHP 7.1.8
```

```
which php
/usr/local/bin/php
```

![screen shot 2017-09-21 at 11 13 01](https://user-images.githubusercontent.com/125499/30688224-e0f9fd1e-9ebd-11e7-9816-bffb802498c7.png)

I also tried setting the Connection Type to other values. When it's set to socket the wheel icon just keeps spinning, any other, nothing happens on Command+Click.

**Comment by tamc** [comment]
I think you need to have the path to the folder where the php binary lives, rather than the path to the php binary itself. i.e., /usr/local/bin not /usr/local/bin/php

**Comment by sr1994lu** [comment]
I confirmed to work!!

i.e. `/Users/user/.anyenv/envs/phpenv/versions/7.2.0RC1/bin`

**Comment by bstrahija** [comment]
Yes it works if you put in the directory path. For me it's:

`/usr/local/bin`

But autocomplete doesn't work at all.

**Comment by sr1994lu** [comment]
@bstrahija What I mean is not `/usr/local/bin`.

Actually php has more executable.
i.e.
```
$ ls /usr/local/Cellar/php71/7.1.8_20/bin
pear pecl phar.phar php-cgi phpize
peardev phar php php-config
```

**Comment by damieng** [comment]
Fixed in v0.6.8 - you now specify the full path to the PHP including the binary name (if it is not in your path or PHP_HOME is not set)

Existing users who set this to a folder will need to change it to the full path.

**Comment by FlorianV54** [comment]
Hi !
what is the correct PHP path on Windows for PHP 7 installed with WAMP (WAMP64 in my case).
If I give this path (C:\wamp64\bin\php\php7.1.9) I have an error

Thanks for your feedback ;)
